spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +364,153 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+                # Get the lesion segmentation mask of the selected 2D axial slice
+                slice_lesion_data = im_lesion_data[sagittal_slice, :, axial_slice]
+                # Get the spinal cord segmentation mask of the selected 2D axial slice
+                slice_sc_data = im_sc_data[sagittal_slice, :, axial_slice]
+                # Get the indices of the lesion mask for the selected axial slice
+                lesion_indices = np.where(slice_lesion_data)[0]
+                # Get the indices of the spinal cord mask for the selected axial slice
+                sc_indices = np.where(slice_sc_data)[0]
+
+                # Compute ventral and dorsal tissue bridges
+                dorsal_bridge_width = lesion_indices[0] - sc_indices[0]         # [0] returns the most dorsal elements
+                # if the lesion extends the spinal cord, the dorsal bridge is set to 0
+                if dorsal_bridge_width < 0:
+                    dorsal_bridge_width = 0
+                ventral_bridge_width = sc_indices[-1] - lesion_indices[-1]      # [-1] returns the most ventral elements
+                # if the lesion extends the spinal cord, the ventral bridge is set to 0
+                if ventral_bridge_width < 0:
+                    ventral_bridge_width = 0
+
+                tissue_bridges_dict[sagittal_slice, axial_slice] = \
+                    {'dorsal_bridge_width': dorsal_bridge_width,
+                     'ventral_bridge_width': ventral_bridge_width}
+
+        # --------------------------------------
+        # Get minimal tissue bridges
+        # --------------------------------------
+        # Convert the dictionary to a DataFrame (for easier manipulation)
+        # 1. Create a MultiIndex from the dictionary keys
+        index = pd.MultiIndex.from_tuples(tissue_bridges_dict.keys(), names=['sagittal_slice', 'axial_slice'])
+        # 2. Create the DataFrame using the MultiIndex and the dictionary values
+        tissue_bridges_df = pd.DataFrame(list(tissue_bridges_dict.values()), index=index)
+        # 3. Reset the index to make 'sagittal_slice' and 'axial_slice' as columns
+        tissue_bridges_df.reset_index(inplace=True)
+
+        # Get slices of minimum dorsal and ventral tissue bridges for each sagittal slice
+        # NOTE: we get minimum because tissue bridges are quantified as the width of spared tissue at the minimum
+        # distance from cerebrospinal fluid to the lesion boundary
+        for sagittal_slice in sagittal_lesion_slices:
+            # Get df for the selected sagittal slice
+            df_temp = tissue_bridges_df[tissue_bridges_df['sagittal_slice'] == sagittal_slice]
+            # Get the axial slices corresponding to the minimum bridge widths
+            min_dorsal_bridge_width_slice = df_temp.loc[df_temp['dorsal_bridge_width'].idxmin(), 'axial_slice']
+            min_ventral_bridge_width_slice = df_temp.loc[df_temp['ventral_bridge_width'].idxmin(), 'axial_slice']
```

Comment:
It looks like you're finding the minimum _before_ doing angle correction, is that ok?

I guess either:
- the angle is almost the same for the whole lesion, in which case this is approximately correct; or
- the angle changes significantly over the length of the lesion, in which case angle correction is flawed anyway.

Reply:
oops! thank you for catching this! we should actually do angle correction first and then find the minimum tissue width!
I fixed the code and described the change in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489/commits/031339784d1a3a2d58df7669ba824a9368c83f1d